### PR TITLE
Initial attempt at 3.0.0 support

### PR DESCRIPTION
**Pull request by kubukoz**
Some things are still missing, but in general I'd say this will be an easy migration.

- [ ] circe-literal
- [ ] refined macros

Tests will fail because circe-literal is currently removed.

### CODE REVIEW

File: oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/common.scala [hunk 1]
```diff
@@ -34,7 +33,8 @@ object common {
   object Scope {
     def of(rawScope: String): Option[Scope] = refineV[ValidScope](rawScope).toOption
 
-    def refine: RefineMPartiallyApplied[Refined, ValidScope] = refineMV[ValidScope]
+    // todo - refined doesn't provide macros for 3.x yet
```
Comment (kubukoz) [inline oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/common.scala:36]: I have an idea for a smol macro of our own for 3.x...

File: build.sbt [hunk 2]
```diff
@@ -89,13 +84,14 @@ val commonDependencies = {
...
-  cats ++ circe ++ sttp ++ refined ++ plugins
+  cats ++ circe ++ sttp ++ refined
 }
 
 val oauth2Dependencies = {
   val testDependencies = Seq(
-    "org.scalatest" %% "scalatest" % Versions.scalaTest,
-    "io.circe" %% "circe-literal" % Versions.circe
+    "org.scalatest" %% "scalatest" % Versions.scalaTest
+    // todo - circe-literal isn't built for 3.x yest
```
Comment (kubukoz) [inline build.sbt:93]: yest xD

**Comment by kubukoz** [comment]
well, this depends on #64 now (sttp hasn't released for 3.0.0 final before the CE3 migration)

**Comment by majk-p** [comment]
Will be implemented in https://github.com/ocadotechnology/sttp-oauth2/issues/172

**Pull request closed without merging**